src/mds/Server.cc

```diff
@@ -736,11 +742,21 @@ void Server::finish_flush_session(Session *session, version_t seq)
 }
 
 void Server::_session_logged(Session *session, uint64_t state_seq, bool open, version_t pv,
-			     interval_set<inodeno_t>& inos, version_t piv)
+			     interval_set<inodeno_t>& inos, version_t piv,
+			     interval_set<inodeno_t>& purge_inos, LogSegment *ls)
 {
-  dout(10) << "_session_logged " << session->info.inst << " state_seq " << state_seq << " " << (open ? "open":"close")
-	   << " " << pv << dendl;
+  dout(10) << "_session_logged " << session->info.inst
+	   << " state_seq " << state_seq
+	   << " " << (open ? "open":"close")
+	   << " " << pv
+	   << " purge_inos : " << purge_inos
+	   << " seq : " << ls->seq << dendl;
 
+  if (purge_inos.size()){
+    ls->purge_inodes.insert(purge_inos);
```

Comment:
move 'ls->purge_inodes.insert(purge_inos)' into MDCache::purge_inodes

Reply:
^ @ukernel you've approved but this is unresolved. Not a problem?

Reply:
we can't do that because of MDCache::start_purge_inodes